packages/react-components/src/components/TextFieldWithMention/mentionTagHelpers.ts

```diff
@@ -0,0 +1,957 @@
+  }
+  if (rangeStart === tag.plainTextBeginIndex && rangeEnd === plainTextEndIndex) {
+    // the whole tag should be removed
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex) + processedChange;
+    plainTextSelectionEndIndex = tag.plainTextBeginIndex + processedChange.length;
+    processedChange = '';
+    lastProcessedHTMLIndex = closeTagIndex + closeTagLength;
+  } else {
+    // only part of the tag should be removed
+    let startChangeDiff = 0;
+    let endChangeDiff = 0;
+    // need to check only rangeStart > tag.plainTextBeginIndex as when rangeStart === tag.plainTextBeginIndex startChangeDiff = 0 and mentionTagLength shouldn't be subtracted
+    if (rangeStart > tag.plainTextBeginIndex) {
+      startChangeDiff = rangeStart - tag.plainTextBeginIndex - mentionTagLength;
+    }
+    endChangeDiff = rangeEnd - tag.plainTextBeginIndex - mentionTagLength;
+    result += htmlText.substring(lastProcessedHTMLIndex, tag.openTagIndex + tag.openTagBody.length + startChangeDiff);
+
+    if (startIndex < tag.plainTextBeginIndex) {
+      // if the change is before the tag, the selection should start from startIndex (rangeStart will be equal to tag.plainTextBeginIndex)
+      plainTextSelectionEndIndex = startIndex + change.length;
+    } else {
+      // if the change is inside the tag, the selection should start with rangeStart
+      plainTextSelectionEndIndex = rangeStart + processedChange.length;
+    }
+    lastProcessedHTMLIndex = tag.openTagIndex + tag.openTagBody.length + endChangeDiff;
+    // processed change should not be changed as it should be added after the tag
+  }
+  return { result, updatedChange: processedChange, htmlIndex: lastProcessedHTMLIndex, plainTextSelectionEndIndex };
+};
+
+/**
+ * Closing tag information
+ *
+ * @private
+ */
+export type ClosingTagInfoResult = {
+  plainTextEndIndex: number;
+  closeTagIndex: number;
+  closeTagLength: number;
+};
+
+/**
+ * Get closing tag information if exists otherwise return information as for self closing tag
+ *
+ * @private
+ * @param tag - Tag data.
+ * @returns Closing tag information for the provided tag.
+ */
+export const getTagClosingTagInfo = (tag: TagData): ClosingTagInfoResult => {
+  let plainTextEndIndex = 0;
+  let closeTagIndex = 0;
+  let closeTagLength = 0;
+  if (tag.plainTextEndIndex !== undefined && tag.closeTagIndex !== undefined) {
+    // close tag exists
+    plainTextEndIndex = tag.plainTextEndIndex;
+    closeTagIndex = tag.closeTagIndex;
+    // tag.tagType.length + </>
+    closeTagLength = tag.tagType.length + 3;
+  } else if (tag.plainTextBeginIndex !== undefined) {
+    // no close tag
+    plainTextEndIndex = tag.plainTextBeginIndex;
+    closeTagIndex = tag.openTagIndex + tag.openTagBody.length;
+    closeTagLength = 0;
+  }
+  return { plainTextEndIndex, closeTagIndex, closeTagLength };
+};
+
+/**
+ * Props for update HTML function
+ *
+ * @private
+ */
+export type UpdateHTMLProps = {
```

Comment:
Given that this is only used explicitly in one place, do we need to define it separately, rather than just inline with the method parameters?

Reply:
As we disscussed previously, separate props types are created to setup parameters correctly (this change was requested in the previous PR)

cc @emlynmac, @Leah-Xia-Microsoft